### PR TITLE
Add FsCheck property-based fuzz tests (addresses Code Scanning alert #15)

### PR DESCRIPTION
**Pull request by alastairlundy**
Adds FsCheck 3.4.0 property-based testing to address GitHub Code Scanning alert #15 (project is not fuzzed).

7 new fuzz test files covering security-critical escapers, tokenizer, builders, and policy classes. All 395 tests pass.

## Summary by CodeRabbit

- **Tests**
  - Added property-based fuzz testing for argument escaping and tokenization, shell escaping, argument specifications, process resource policies, timeout policies, and version parsing.
  - Expanded coverage for edge cases, invalid inputs, exception handling, default values, equality behavior, escaping rules, and output consistency.
  - Added centrally managed FsCheck test support for more consistent and thorough validation.

### CODE REVIEW

File: tests/CliInvoke.Tests/Fuzzing/ArgumentTokenizerFuzzTests.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            })
+            .QuickCheckThrowOnFailure();
+    }
+
+    [Test]
+    public void Tokenize_SpaceSeparatedWords_ProducesCorrectCount()
+    {
+        Prop.ForAll<string>(value =>
+            {
+                if (string.IsNullOrWhiteSpace(value) || value.Any(char.IsWhiteSpace) || value.Contains('"'))
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Fuzzing/ArgumentTokenizerFuzzTests.cs]: **Accepted**. Updated both properties to use generated inputs:

1. Tokenize_SpaceSeparatedWords_ProducesCorrectCount now constructs inputs from two generated non-quoted words separated by a space, ensuring the property actually tests multiple-token behavior and asserts both token count and values.

2. Tokenize_DoubleQuoteInsideQuotes_ProducesLiteralQuote now uses the generated value parameter to construct varied inputs (pattern: "value""value"), verifying both token count and the expected escaped output.

Tests pass with these changes.

File: tests/CliInvoke.Tests/Fuzzing/VersionParseExtensionsFuzzTests.cs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+                try
+                {
+                    Version result = Version.GracefulParse(value);
+                    return result.Major >= 0;
+                }
+                catch (ArgumentException)
+                {
+                    return true;
+                }
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Fuzzing/VersionParseExtensionsFuzzTests.cs]: Applied. Strengthened test properties to remove overly permissive exception handling and compare parsed components against generated segment values. Verified all tests pass (395/395).

File: tests/CliInvoke.Tests/Fuzzing/ShellArgumentEscaperFuzzTests.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    [Test]
+    public void EscapeForPowerShell_OutputContainsNoUnescapedMetacharacters()
+    {
+        Prop.ForAll<string>(value =>
+            {
+                if (string.IsNullOrEmpty(value) || value.Contains('\n') || value.Contains('\r'))
+                    return true;
+
+                string escaped = ShellArgumentEscaper.EscapeForPowerShell(value);
+                return escaped.Length >= value.Length;
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Fuzzing/ShellArgumentEscaperFuzzTests.cs]: **Applied.** The tests now validate that each PowerShell metacharacter is escaped with backtick (`) and each CMD metacharacter is escaped with caret (^), with double quotes escaped as "". This catches regressions where the escaper might return input unchanged, which the length check would have missed.

File: tests/CliInvoke.Tests/Fuzzing/ProcessResourcePolicyFuzzTests.cs [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+    [Test]
+    public void SetMaxWorkingSet_LessThanMin_ThrowsArgumentOutOfRangeException()
+    {
+        Prop.ForAll<int>(max =>
+                {
+                    if (max <= 1) return true;
+
+                    var spec = new ProcessResourcePolicySpec();
+                    spec.SetMinWorkingSet((nint)(max + 1));
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Fuzzing/ProcessResourcePolicyFuzzTests.cs:52]: Good catch. The guard clauses only checked lower bounds, so FsCheck could generate `int.MaxValue` and the `+ 1` would silently overflow, making the test pass trivially. I've added an upper-bound guard (`>= int.MaxValue - 1`) in all three locations so the generated values always produce a valid, larger positive result. All 395 tests pass.

File: tests/CliInvoke.Tests/Fuzzing/ArgumentsSpecFuzzTests.cs [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+    [Test]
+    public async Task AddEnumerable_WithEmptyValues_ThrowsArgumentException()
+    {
+        var spec = new ArgumentsSpec();
+
+        try
+        {
+            spec.AddEnumerable(Enumerable.Empty<string>(), escape: false);
+            return;
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Fuzzing/ArgumentsSpecFuzzTests.cs]: Good catch. Replaced the try/catch/return pattern with TUnit's `Assert.That(() => ...).Throws<T>()` in all three tests so they now fail when the expected exception is absent. All 395 tests pass.

**Comment by alastairlundy** [comment]
@coderabbitai review

File: tests/CliInvoke.Tests/Fuzzing/ShellArgumentEscaperFuzzTests.cs [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+ 
+                // Verify each PowerShell metacharacter is properly escaped
+                foreach (char metachar in PowerShellMetacharacters)
+                {
+                    if (value.Contains(metachar))
+                    {
+                        // The escaped sequence should be backtick followed by the metachar
+                        string expectedEscaped = $"`{metachar}";
+                        if (!escaped.Contains(expectedEscaped))
+                            return false;
```
[the remark this answers is not in the document]
Comment (alastairlundy) [reply tests/CliInvoke.Tests/Fuzzing/ShellArgumentEscaperFuzzTests.cs]: Good catch. The Contains-based checks could indeed pass incorrect escaping where only one instance of a repeated metacharacter was escaped. Replaced both the PowerShell and CMD validations with character-by-character expected-output comparison that mirrors the escaper logic. All 395 tests pass.
